src/ctapipe/image/muon/fitting.py

```diff
@@ -75,52 +143,137 @@ def taubin_circle_fit(x, y, mask):
     x_masked = x[mask]
     y_masked = y[mask]
 
-    R = x.max()  # x.max() just happens to be identical with R in many cases.
+    max_fov = 2 * x.max()
+
+    if weights is None:
+        weights_masked = np.ones(len(x_masked))
+    else:
+        weights_masked = weights[mask]
+
+    if original_unit == u.deg:
```

Comment:
lines 153-156 can be simplified to this:

```suggestion
    if original_unit.is_equivalent(u.deg):
        r_initial = (1.1 * u.deg).to(original_unit) if r_initial is None else r_initial
```